src/backend/utils/init/postinit.c

```diff
@@ -652,14 +652,17 @@ InitPostgres(const char *in_dbname, Oid dboid, const char *username,
 	GPMemoryProtect_Init();
 
 #ifdef USE_ORCA
-	/* Initialize GPOPT */
-	OptimizerMemoryContext = AllocSetContextCreate(TopMemoryContext,
-												   "GPORCA Top-level Memory Context",
-												   ALLOCSET_DEFAULT_MINSIZE,
-												   ALLOCSET_DEFAULT_INITSIZE,
-												   ALLOCSET_DEFAULT_MAXSIZE);
-
-	InitGPOPT();
+	if (GP_ROLE_DISPATCH == Gp_role)
```

Comment:
```suggestion
	if (Gp_role == GP_ROLE_DISPATCH )
```

LGTM, but only one question. Why did you use this unusual notation? As for me, it's very counterintuitive. Moreover, there is only one place with such order that you used to copy from. I vote to make order consistent in _your patch_ with most of the codebase.

Reply:
Yes you are right. I did this out of habit, as in an ORCA. Corrected.